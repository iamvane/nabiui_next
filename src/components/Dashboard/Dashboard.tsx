import * as React from 'react';
import { connect } from 'react-redux';
import {
  Action,
  Dispatch
} from 'redux';
import { RouteComponentProps } from 'react-router';
import Link from 'next/link';

import {
  CircularProgress,
  Typography } from '@material-ui/core';

import { StoreState } from '../../redux/reducers/store';
import { UserType } from '../../redux/models/UserModel';
import { fetchUser } from '../../redux/actions/UserActions';
import { InstructorType } from '../../redux/models/InstructorModel';
import { page } from '../../utils/analytics';
import SnackBar from '../common/SnackBar';
import SectionTitle from '../common/SectionTitle';
import { AnnouncementConstants } from '../common/constants/Announcement';
import { LoggedInPageTemplate } from '../common/Templates/LoggedInPageTemplate';
import { Routes } from '../common/constants/Routes';
import { Role } from '../Auth/Registration/constants';
import InviteFriends from '../InviteFriends/InviteFriends';
import InstructorDashboard from './InstructorDashboard/InstructorDashboard';
import ParentStudentDashboard from './ParentStudentDashboard/ParentStudentDashboard';
import {
  DashboardComponent,
  PreLaunchInstructorDashboardComponent as constants
} from './constants';

interface State {
  showSnackbar: boolean;
}

interface StateProps {
  user: UserType;
  isRequesting: boolean;
  firstName: string;
}

interface DispatchProps {
  fetchUser: () => void;
}

interface OwnProps {
}

interface Props extends
  OwnProps,
  RouteComponentProps<{}>,
  StateProps,
  DispatchProps {}

export class Dashboard extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = {
      showSnackbar: false
    };
  }

  public async componentDidMount(): Promise<void> {
    await this.props.fetchUser();

    // if (this.props.location.state && this.props.location.state.redirectedFrom === Routes.BuildRequest) {
    //   this.setState({
    //     showSnackbar: true
    //   });
    // }

    const userId = this.props.user ? this.props.user.email : 'anonymous';

    const analiticsProps = {
      userId,
      properties: {
        referrer: document.referrer
      }
    };
    page('Dashboard', analiticsProps);
  }

  public closeSnackbar = () => this.setState({ showSnackbar: false });

  render() {
    const LinkedInPic = 'https://nabimusic.s3.us-east-2.amazonaws.com/assets/images/linkedin-pic.jpg';
    return (
      <React.Fragment>
        {this.props.isRequesting ?
        <div className="nabi-text-center">
          <CircularProgress />
        </div> :
        <React.Fragment>
          <LoggedInPageTemplate
            sidebarContent={
              <InviteFriends />
            }
            mainContent={
              this.props.user.role === Role.instructor ?
              <InstructorDashboard user={this.props.user} /> : <ParentStudentDashboard role={this.props.user.role} />}
            pageTitle={DashboardComponent.pageTitle}
          />
          <SnackBar
            isOpen={this.state.showSnackbar}
            message={AnnouncementConstants.requestSentMessage}
            handleClose={this.closeSnackbar}
            variant="success"
            hideIcon={true}
          />
        </React.Fragment>}
      </React.Fragment>
    );
  }
}

function mapStateToProps(state: StoreState, _ownProps: OwnProps): StateProps {
  const {
    actions: {
      fetchUser: {
        isRequesting,
      }
    }
  } = state.user;

  return {
    user: state.user.user,
    firstName: state.user.user.firstName,
    isRequesting
  };
}

const mapDispatchToProps = (
  dispatch: Dispatch<Action>
): DispatchProps => ({
  fetchUser: () => dispatch(fetchUser())
});

export default connect(mapStateToProps, mapDispatchToProps)(Dashboard);
