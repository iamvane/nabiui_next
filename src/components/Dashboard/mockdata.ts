export const studentMockDataEmpty = [];

export const studentMockData = [
  {
    user: {
      id: '0',
      email: 'rox@gmail.com',
      role: 'student',
      firstName: 'Roxana',
      lastName: 'Gallardo',
      birthday: '1996-07-12T04:00:00.000Z'
    },
    students: [
      {
        name: 'Roxana',
        age: 22,
        instrument: 'piano',
        skillLevel: 'beginner',
        placeForLessons: 'home',
        lessonDuration: '45 mins'
      }
    ],
    lessonsRemaining: 4
  },
  {
    user: {
      id: '0',
      email: 'kev@gmail.com',
      role: 'student',
      firstName: 'Kevin',
      lastName: 'Charles',
      birthday: '1992-05-26T04:00:00.000Z'
    },
    students: [
      {
        name: 'Kevin',
        age: 22,
        instrument: 'piano',
        skillLevel: 'beginner',
        placeForLessons: 'home',
        lessonDuration: '45 mins'
      }
    ],
    lessonsRemaining: 0
  },
  {
    user: {
      id: '0',
      email: 'sam@gmail.com',
      role: 'parent',
      firstName: 'Sam',
      lastName: 'Flood',
      birthday: '1960-05-26T04:00:00.000Z'
    },
    students: [
      {
        name: 'Zoe',
        age: 6,
        instrument: 'piano',
        skillLevel: 'beginner',
        placeForLessons: 'home',
        lessonDuration: '45 mins'
      },
      {
        name: 'Esme',
        age: 9,
        instrument: 'piano',
        skillLevel: 'beginner',
        placeForLessons: 'home',
        lessonDuration: '45 mins'
      }
    ],
    lessonsRemaining: 5
  },
];
