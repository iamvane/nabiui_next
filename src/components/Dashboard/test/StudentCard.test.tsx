import * as React from 'react';
import { shallow } from 'enzyme';

import StudentCard from 'components/Dashboard/InstructorDashboard/StudentCard';
import { studentMockData } from 'components/Dashboard/mockdata';

describe('StudentCard', () => {
  const  wrapper = shallow(
    <StudentCard
      student={studentMockData[0]}
      toggleBuyLessonsForm={jest.fn()}
    />
  );

  it('Matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
