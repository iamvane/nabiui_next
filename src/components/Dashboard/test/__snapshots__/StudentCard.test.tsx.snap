// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StudentCard Matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StudentCard
    student={
      Object {
        "lessonsRemaining": 4,
        "students": Array [
          Object {
            "age": 22,
            "instrument": "piano",
            "lessonDuration": "45 mins",
            "name": "Roxana",
            "placeForLessons": "home",
            "skillLevel": "beginner",
          },
        ],
        "user": Object {
          "birthday": "1996-07-12T04:00:00.000Z",
          "email": "rox@gmail.com",
          "firstName": "Roxana",
          "id": "0",
          "lastName": "Gallardo",
          "role": "student",
        },
      }
    }
    toggleBuyLessonsForm={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <WithStyles(Grid)
          item={true}
          md={5}
          xs={12}
        >
          <WithStyles(Grid)
            container={true}
          >
            <WithStyles(Grid)
              item={true}
              xs={6}
            >
              <WithStyles(Typography)>
                Student
              </WithStyles(Typography)>
            </WithStyles(Grid)>
            <WithStyles(Grid)
              className="nabi-text-right"
              item={true}
              xs={6}
            >
              <WithStyles(Typography)>
                Roxana
              </WithStyles(Typography)>
            </WithStyles(Grid)>
            
            <WithStyles(Grid)
              item={true}
              xs={6}
            >
              <WithStyles(Typography)>
                Age
              </WithStyles(Typography)>
            </WithStyles(Grid)>
            <WithStyles(Grid)
              className="nabi-text-right"
              item={true}
              xs={6}
            >
              <WithStyles(Typography)>
                22
              </WithStyles(Typography)>
            </WithStyles(Grid)>
            <WithStyles(Grid)
              item={true}
              xs={6}
            >
              <WithStyles(Typography)>
                Instrument
              </WithStyles(Typography)>
            </WithStyles(Grid)>
            <WithStyles(Grid)
              className="nabi-text-right"
              item={true}
              xs={6}
            >
              <WithStyles(Typography)>
                Piano
              </WithStyles(Typography)>
            </WithStyles(Grid)>
            <WithStyles(Grid)
              item={true}
              xs={6}
            >
              <WithStyles(Typography)>
                Skill level
              </WithStyles(Typography)>
            </WithStyles(Grid)>
            <WithStyles(Grid)
              className="nabi-text-right"
              item={true}
              xs={6}
            >
              <WithStyles(Typography)>
                beginner
              </WithStyles(Typography)>
            </WithStyles(Grid)>
          </WithStyles(Grid)>
        </WithStyles(Grid)>,
        <WithStyles(Grid)
          className="nabi-text-center nabi-text-left-md nabi-padding-left-small-md"
          item={true}
          md={5}
          xs={12}
        >
          <WithStyles(Button)
            className="nabi-responsive-button nabi-margin-top-xsmall"
            color="primary"
            onClick={[MockFunction]}
            variant="contained"
          >
            4 Lessons Remaining
          </WithStyles(Button)>
          <br />
          <WithStyles(Button)
            className="nabi-responsive-button nabi-margin-top-xsmall nabi-margin-right-xsmall"
            color="primary"
            variant="contained"
          >
            Grade Lesson
          </WithStyles(Button)>
          <WithStyles(Button)
            className="nabi-responsive-button nabi-margin-top-xsmall"
            color="primary"
            variant="contained"
          >
            Contact Student
          </WithStyles(Button)>
        </WithStyles(Grid)>,
      ],
      "className": "nabi-margin-bottom-small",
      "container": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <WithStyles(Grid)
            container={true}
          >
            <WithStyles(Grid)
              item={true}
              xs={6}
            >
              <WithStyles(Typography)>
                Student
              </WithStyles(Typography)>
            </WithStyles(Grid)>
            <WithStyles(Grid)
              className="nabi-text-right"
              item={true}
              xs={6}
            >
              <WithStyles(Typography)>
                Roxana
              </WithStyles(Typography)>
            </WithStyles(Grid)>
            
            <WithStyles(Grid)
              item={true}
              xs={6}
            >
              <WithStyles(Typography)>
                Age
              </WithStyles(Typography)>
            </WithStyles(Grid)>
            <WithStyles(Grid)
              className="nabi-text-right"
              item={true}
              xs={6}
            >
              <WithStyles(Typography)>
                22
              </WithStyles(Typography)>
            </WithStyles(Grid)>
            <WithStyles(Grid)
              item={true}
              xs={6}
            >
              <WithStyles(Typography)>
                Instrument
              </WithStyles(Typography)>
            </WithStyles(Grid)>
            <WithStyles(Grid)
              className="nabi-text-right"
              item={true}
              xs={6}
            >
              <WithStyles(Typography)>
                Piano
              </WithStyles(Typography)>
            </WithStyles(Grid)>
            <WithStyles(Grid)
              item={true}
              xs={6}
            >
              <WithStyles(Typography)>
                Skill level
              </WithStyles(Typography)>
            </WithStyles(Grid)>
            <WithStyles(Grid)
              className="nabi-text-right"
              item={true}
              xs={6}
            >
              <WithStyles(Typography)>
                beginner
              </WithStyles(Typography)>
            </WithStyles(Grid)>
          </WithStyles(Grid)>,
          "item": true,
          "md": 5,
          "xs": 12,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(Grid)
                item={true}
                xs={6}
              >
                <WithStyles(Typography)>
                  Student
                </WithStyles(Typography)>
              </WithStyles(Grid)>,
              <WithStyles(Grid)
                className="nabi-text-right"
                item={true}
                xs={6}
              >
                <WithStyles(Typography)>
                  Roxana
                </WithStyles(Typography)>
              </WithStyles(Grid)>,
              "",
              <WithStyles(Grid)
                item={true}
                xs={6}
              >
                <WithStyles(Typography)>
                  Age
                </WithStyles(Typography)>
              </WithStyles(Grid)>,
              <WithStyles(Grid)
                className="nabi-text-right"
                item={true}
                xs={6}
              >
                <WithStyles(Typography)>
                  22
                </WithStyles(Typography)>
              </WithStyles(Grid)>,
              <WithStyles(Grid)
                item={true}
                xs={6}
              >
                <WithStyles(Typography)>
                  Instrument
                </WithStyles(Typography)>
              </WithStyles(Grid)>,
              <WithStyles(Grid)
                className="nabi-text-right"
                item={true}
                xs={6}
              >
                <WithStyles(Typography)>
                  Piano
                </WithStyles(Typography)>
              </WithStyles(Grid)>,
              <WithStyles(Grid)
                item={true}
                xs={6}
              >
                <WithStyles(Typography)>
                  Skill level
                </WithStyles(Typography)>
              </WithStyles(Grid)>,
              <WithStyles(Grid)
                className="nabi-text-right"
                item={true}
                xs={6}
              >
                <WithStyles(Typography)>
                  beginner
                </WithStyles(Typography)>
              </WithStyles(Grid)>,
            ],
            "container": true,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <WithStyles(Typography)>
                  Student
                </WithStyles(Typography)>,
                "item": true,
                "xs": 6,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Student",
                },
                "ref": null,
                "rendered": "Student",
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <WithStyles(Typography)>
                  Roxana
                </WithStyles(Typography)>,
                "className": "nabi-text-right",
                "item": true,
                "xs": 6,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Roxana",
                },
                "ref": null,
                "rendered": "Roxana",
                "type": [Function],
              },
              "type": [Function],
            },
            "",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <WithStyles(Typography)>
                  Age
                </WithStyles(Typography)>,
                "item": true,
                "xs": 6,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Age",
                },
                "ref": null,
                "rendered": "Age",
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <WithStyles(Typography)>
                  22
                </WithStyles(Typography)>,
                "className": "nabi-text-right",
                "item": true,
                "xs": 6,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": 22,
                },
                "ref": null,
                "rendered": 22,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <WithStyles(Typography)>
                  Instrument
                </WithStyles(Typography)>,
                "item": true,
                "xs": 6,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Instrument",
                },
                "ref": null,
                "rendered": "Instrument",
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <WithStyles(Typography)>
                  Piano
                </WithStyles(Typography)>,
                "className": "nabi-text-right",
                "item": true,
                "xs": 6,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Piano",
                },
                "ref": null,
                "rendered": "Piano",
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <WithStyles(Typography)>
                  Skill level
                </WithStyles(Typography)>,
                "item": true,
                "xs": 6,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Skill level",
                },
                "ref": null,
                "rendered": "Skill level",
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <WithStyles(Typography)>
                  beginner
                </WithStyles(Typography)>,
                "className": "nabi-text-right",
                "item": true,
                "xs": 6,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "beginner",
                },
                "ref": null,
                "rendered": "beginner",
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(Button)
              className="nabi-responsive-button nabi-margin-top-xsmall"
              color="primary"
              onClick={[MockFunction]}
              variant="contained"
            >
              4 Lessons Remaining
            </WithStyles(Button)>,
            <br />,
            <WithStyles(Button)
              className="nabi-responsive-button nabi-margin-top-xsmall nabi-margin-right-xsmall"
              color="primary"
              variant="contained"
            >
              Grade Lesson
            </WithStyles(Button)>,
            <WithStyles(Button)
              className="nabi-responsive-button nabi-margin-top-xsmall"
              color="primary"
              variant="contained"
            >
              Contact Student
            </WithStyles(Button)>,
          ],
          "className": "nabi-text-center nabi-text-left-md nabi-padding-left-small-md",
          "item": true,
          "md": 5,
          "xs": 12,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "4 Lessons Remaining",
              "className": "nabi-responsive-button nabi-margin-top-xsmall",
              "color": "primary",
              "onClick": [MockFunction],
              "variant": "contained",
            },
            "ref": null,
            "rendered": "4 Lessons Remaining",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Grade Lesson",
              "className": "nabi-responsive-button nabi-margin-top-xsmall nabi-margin-right-xsmall",
              "color": "primary",
              "variant": "contained",
            },
            "ref": null,
            "rendered": "Grade Lesson",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Contact Student",
              "className": "nabi-responsive-button nabi-margin-top-xsmall",
              "color": "primary",
              "variant": "contained",
            },
            "ref": null,
            "rendered": "Contact Student",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <WithStyles(Grid)
            item={true}
            md={5}
            xs={12}
          >
            <WithStyles(Grid)
              container={true}
            >
              <WithStyles(Grid)
                item={true}
                xs={6}
              >
                <WithStyles(Typography)>
                  Student
                </WithStyles(Typography)>
              </WithStyles(Grid)>
              <WithStyles(Grid)
                className="nabi-text-right"
                item={true}
                xs={6}
              >
                <WithStyles(Typography)>
                  Roxana
                </WithStyles(Typography)>
              </WithStyles(Grid)>
              
              <WithStyles(Grid)
                item={true}
                xs={6}
              >
                <WithStyles(Typography)>
                  Age
                </WithStyles(Typography)>
              </WithStyles(Grid)>
              <WithStyles(Grid)
                className="nabi-text-right"
                item={true}
                xs={6}
              >
                <WithStyles(Typography)>
                  22
                </WithStyles(Typography)>
              </WithStyles(Grid)>
              <WithStyles(Grid)
                item={true}
                xs={6}
              >
                <WithStyles(Typography)>
                  Instrument
                </WithStyles(Typography)>
              </WithStyles(Grid)>
              <WithStyles(Grid)
                className="nabi-text-right"
                item={true}
                xs={6}
              >
                <WithStyles(Typography)>
                  Piano
                </WithStyles(Typography)>
              </WithStyles(Grid)>
              <WithStyles(Grid)
                item={true}
                xs={6}
              >
                <WithStyles(Typography)>
                  Skill level
                </WithStyles(Typography)>
              </WithStyles(Grid)>
              <WithStyles(Grid)
                className="nabi-text-right"
                item={true}
                xs={6}
              >
                <WithStyles(Typography)>
                  beginner
                </WithStyles(Typography)>
              </WithStyles(Grid)>
            </WithStyles(Grid)>
          </WithStyles(Grid)>,
          <WithStyles(Grid)
            className="nabi-text-center nabi-text-left-md nabi-padding-left-small-md"
            item={true}
            md={5}
            xs={12}
          >
            <WithStyles(Button)
              className="nabi-responsive-button nabi-margin-top-xsmall"
              color="primary"
              onClick={[MockFunction]}
              variant="contained"
            >
              4 Lessons Remaining
            </WithStyles(Button)>
            <br />
            <WithStyles(Button)
              className="nabi-responsive-button nabi-margin-top-xsmall nabi-margin-right-xsmall"
              color="primary"
              variant="contained"
            >
              Grade Lesson
            </WithStyles(Button)>
            <WithStyles(Button)
              className="nabi-responsive-button nabi-margin-top-xsmall"
              color="primary"
              variant="contained"
            >
              Contact Student
            </WithStyles(Button)>
          </WithStyles(Grid)>,
        ],
        "className": "nabi-margin-bottom-small",
        "container": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(Grid)
              container={true}
            >
              <WithStyles(Grid)
                item={true}
                xs={6}
              >
                <WithStyles(Typography)>
                  Student
                </WithStyles(Typography)>
              </WithStyles(Grid)>
              <WithStyles(Grid)
                className="nabi-text-right"
                item={true}
                xs={6}
              >
                <WithStyles(Typography)>
                  Roxana
                </WithStyles(Typography)>
              </WithStyles(Grid)>
              
              <WithStyles(Grid)
                item={true}
                xs={6}
              >
                <WithStyles(Typography)>
                  Age
                </WithStyles(Typography)>
              </WithStyles(Grid)>
              <WithStyles(Grid)
                className="nabi-text-right"
                item={true}
                xs={6}
              >
                <WithStyles(Typography)>
                  22
                </WithStyles(Typography)>
              </WithStyles(Grid)>
              <WithStyles(Grid)
                item={true}
                xs={6}
              >
                <WithStyles(Typography)>
                  Instrument
                </WithStyles(Typography)>
              </WithStyles(Grid)>
              <WithStyles(Grid)
                className="nabi-text-right"
                item={true}
                xs={6}
              >
                <WithStyles(Typography)>
                  Piano
                </WithStyles(Typography)>
              </WithStyles(Grid)>
              <WithStyles(Grid)
                item={true}
                xs={6}
              >
                <WithStyles(Typography)>
                  Skill level
                </WithStyles(Typography)>
              </WithStyles(Grid)>
              <WithStyles(Grid)
                className="nabi-text-right"
                item={true}
                xs={6}
              >
                <WithStyles(Typography)>
                  beginner
                </WithStyles(Typography)>
              </WithStyles(Grid)>
            </WithStyles(Grid)>,
            "item": true,
            "md": 5,
            "xs": 12,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <WithStyles(Grid)
                  item={true}
                  xs={6}
                >
                  <WithStyles(Typography)>
                    Student
                  </WithStyles(Typography)>
                </WithStyles(Grid)>,
                <WithStyles(Grid)
                  className="nabi-text-right"
                  item={true}
                  xs={6}
                >
                  <WithStyles(Typography)>
                    Roxana
                  </WithStyles(Typography)>
                </WithStyles(Grid)>,
                "",
                <WithStyles(Grid)
                  item={true}
                  xs={6}
                >
                  <WithStyles(Typography)>
                    Age
                  </WithStyles(Typography)>
                </WithStyles(Grid)>,
                <WithStyles(Grid)
                  className="nabi-text-right"
                  item={true}
                  xs={6}
                >
                  <WithStyles(Typography)>
                    22
                  </WithStyles(Typography)>
                </WithStyles(Grid)>,
                <WithStyles(Grid)
                  item={true}
                  xs={6}
                >
                  <WithStyles(Typography)>
                    Instrument
                  </WithStyles(Typography)>
                </WithStyles(Grid)>,
                <WithStyles(Grid)
                  className="nabi-text-right"
                  item={true}
                  xs={6}
                >
                  <WithStyles(Typography)>
                    Piano
                  </WithStyles(Typography)>
                </WithStyles(Grid)>,
                <WithStyles(Grid)
                  item={true}
                  xs={6}
                >
                  <WithStyles(Typography)>
                    Skill level
                  </WithStyles(Typography)>
                </WithStyles(Grid)>,
                <WithStyles(Grid)
                  className="nabi-text-right"
                  item={true}
                  xs={6}
                >
                  <WithStyles(Typography)>
                    beginner
                  </WithStyles(Typography)>
                </WithStyles(Grid)>,
              ],
              "container": true,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <WithStyles(Typography)>
                    Student
                  </WithStyles(Typography)>,
                  "item": true,
                  "xs": 6,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Student",
                  },
                  "ref": null,
                  "rendered": "Student",
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <WithStyles(Typography)>
                    Roxana
                  </WithStyles(Typography)>,
                  "className": "nabi-text-right",
                  "item": true,
                  "xs": 6,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Roxana",
                  },
                  "ref": null,
                  "rendered": "Roxana",
                  "type": [Function],
                },
                "type": [Function],
              },
              "",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <WithStyles(Typography)>
                    Age
                  </WithStyles(Typography)>,
                  "item": true,
                  "xs": 6,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Age",
                  },
                  "ref": null,
                  "rendered": "Age",
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <WithStyles(Typography)>
                    22
                  </WithStyles(Typography)>,
                  "className": "nabi-text-right",
                  "item": true,
                  "xs": 6,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": 22,
                  },
                  "ref": null,
                  "rendered": 22,
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <WithStyles(Typography)>
                    Instrument
                  </WithStyles(Typography)>,
                  "item": true,
                  "xs": 6,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Instrument",
                  },
                  "ref": null,
                  "rendered": "Instrument",
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <WithStyles(Typography)>
                    Piano
                  </WithStyles(Typography)>,
                  "className": "nabi-text-right",
                  "item": true,
                  "xs": 6,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Piano",
                  },
                  "ref": null,
                  "rendered": "Piano",
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <WithStyles(Typography)>
                    Skill level
                  </WithStyles(Typography)>,
                  "item": true,
                  "xs": 6,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Skill level",
                  },
                  "ref": null,
                  "rendered": "Skill level",
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <WithStyles(Typography)>
                    beginner
                  </WithStyles(Typography)>,
                  "className": "nabi-text-right",
                  "item": true,
                  "xs": 6,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "beginner",
                  },
                  "ref": null,
                  "rendered": "beginner",
                  "type": [Function],
                },
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(Button)
                className="nabi-responsive-button nabi-margin-top-xsmall"
                color="primary"
                onClick={[MockFunction]}
                variant="contained"
              >
                4 Lessons Remaining
              </WithStyles(Button)>,
              <br />,
              <WithStyles(Button)
                className="nabi-responsive-button nabi-margin-top-xsmall nabi-margin-right-xsmall"
                color="primary"
                variant="contained"
              >
                Grade Lesson
              </WithStyles(Button)>,
              <WithStyles(Button)
                className="nabi-responsive-button nabi-margin-top-xsmall"
                color="primary"
                variant="contained"
              >
                Contact Student
              </WithStyles(Button)>,
            ],
            "className": "nabi-text-center nabi-text-left-md nabi-padding-left-small-md",
            "item": true,
            "md": 5,
            "xs": 12,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "4 Lessons Remaining",
                "className": "nabi-responsive-button nabi-margin-top-xsmall",
                "color": "primary",
                "onClick": [MockFunction],
                "variant": "contained",
              },
              "ref": null,
              "rendered": "4 Lessons Remaining",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Grade Lesson",
                "className": "nabi-responsive-button nabi-margin-top-xsmall nabi-margin-right-xsmall",
                "color": "primary",
                "variant": "contained",
              },
              "ref": null,
              "rendered": "Grade Lesson",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Contact Student",
                "className": "nabi-responsive-button nabi-margin-top-xsmall",
                "color": "primary",
                "variant": "contained",
              },
              "ref": null,
              "rendered": "Contact Student",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
